src/main/scala/coreplex/RocketCoreplex.scala

```diff
@@ -95,67 +96,72 @@ trait HasRocketTiles extends HasTiles
-      case SharedMemoryTLEdge => sharedMemoryTLEdge
-      case RocketCrossingKey => List(crossing)
-    }
-
-    val wrapper = crossing.crossingType match {
-      case SynchronousCrossing(params) => {
-        val wrapper = LazyModule(new SyncRocketTile(tp)(pWithExtra))
-        sbus.fromSyncTiles(params, crossing.master.adapterChain(this), tp.name) :=* wrapper.masterNode
-        FlipRendering { implicit p => wrapper.slaveNode :*= pbus.toSyncSlaves(crossing.slave.adapterChain(this), tp.name) }
-        wrapper
-      }
-      case AsynchronousCrossing(depth, sync) => {
-        val wrapper = LazyModule(new AsyncRocketTile(tp)(pWithExtra))
-        sbus.fromAsyncTiles(depth, sync, crossing.master.adapterChain(this), tp.name) :=* wrapper.masterNode
-        FlipRendering { implicit p => wrapper.slaveNode :*= pbus.toAsyncSlaves(sync, crossing.slave.adapterChain(this), tp.name) }
-        wrapper
-      }
-      case RationalCrossing(direction) => {
-        val wrapper = LazyModule(new RationalRocketTile(tp)(pWithExtra))
-        sbus.fromRationalTiles(direction, crossing.master.adapterChain(this), tp.name) :=* wrapper.masterNode
-        FlipRendering { implicit p => wrapper.slaveNode :*= pbus.toRationalSlaves(crossing.slave.adapterChain(this), tp.name) }
-        wrapper
-      }
-    }
-    tp.name.foreach(wrapper.suggestName) // Try to stabilize this name for downstream tools
-
-    // Local Interrupts must be synchronized to the core clock
-    // before being passed into this module.
-    // This allows faster latency for interrupts which are already synchronized.
-    // The CLINT and PLIC outputs interrupts that are synchronous to the periphery clock,
-    // so may or may not need to be synchronized depending on the Tile's crossing type.
-    // Debug interrupt is definitely asynchronous in all cases.
-    val asyncIntXbar  = LazyModule(new IntXbar)
-    asyncIntXbar.suggestName("asyncIntXbar")
-    asyncIntXbar.intnode  := debug.intnode                  // debug
-    wrapper.asyncIntNode  := asyncIntXbar.intnode
-
-    val periphIntXbar = LazyModule(new IntXbar)
-    periphIntXbar.suggestName("periphIntXbar")
-    periphIntXbar.intnode := clint.intnode                  // msip+mtip
-    periphIntXbar.intnode := plic.intnode                   // meip
+    // For legacy reasons, it is convenient to store some state
+    // in the global Parameters about the specific tile being built now
+    val wrapper = LazyModule(new RocketTileWrapper(
+      params = tp,
+      crossing = crossing.crossingType,
+      boundaryBuffers = crossing.boundaryBuffers
+      )(p.alterPartial {
+        case TileKey => tp
+        case BuildRoCC => tp.rocc
+        case SharedMemoryTLEdge => sharedMemoryTLEdge
+        case RocketCrossingKey => List(crossing)
+      })
+    ).suggestName(tp.name)
+
+    // Connect the master ports of the tile to the system bus
+    sbus.fromTile(
+      adapt = {x: TLOutwardNode => wrapper.cross(x) } andThen
```

Comment:
@terpstra Sadly I needed to be explicit to the type checker here, otherwise it couldn't decide which version of `wrapper.cross(...)` applied. If you can come up with a concise syntax that's cool, otherwise I'd rather just write `wrapper.crossTL _` than go through the extra steps.